ChemistryLib/PhreeqcIOData/AqueousSolution.h

```diff
@@ -24,10 +27,16 @@ namespace PhreeqcIOData
 {
 struct Component
 {
-    explicit Component(std::string name_) : name(std::move(name_)) {}
+    explicit Component(std::string name_,
+                       std::size_t const num_chemical_systems_)
+        : name(std::move(name_)),
+          amount(MathLib::MatrixVectorTraits<GlobalVector>::newInstance(
+              num_chemical_systems_))
+    {
+    }
 
     std::string const name;
-    double amount = std::numeric_limits<double>::quiet_NaN();
+    std::unique_ptr<GlobalVector> amount;
```

Comment:
Maybe `amount` is a too general term. If you have something more specific, would be great.

Reply:
You will see this change in the next a few pull requests. :smile: